layers/manipulators/layerops.py

```diff
@@ -0,0 +1,344 @@
+# Example Use:
+# from layers.manipulators.layerops import LayerOps
+# from layers.core.layer import Layer
+
+# demo = Layer()
+# demo.load_file("C:\Users\attack\Downloads\layer.json")
+# demo2 = Layer()
+# demo2.load_input(Existing_Layer_String_Previously_Loaded)
+
+# lo = LayerOps(score=lambda x: x[0] * x[1], name=lambda x: x[1],
+#               desc=lambda x: "This is an list example")
+# out_layer = lo.process([demo, demo2])
+# out_layer.export_file("C:\demo_layer1.json")
+
+# lo2 = LayerOps(score=lambda x: x['a'], color=lambda x: x['b'],
+#                desc=lambda x: "This is a dict example")
+# out_layer2 = lo2.process({'a': demo, 'b': demo2})
+# dict_layer = out_layer2.get_dict()
+# print(dict_layer)
+# out_layer2.export_file("C:\demo_layer2.json")
+
+import copy
+from layers.core import Layer
+
+
+class InvalidFormat(Exception):
+    pass
+
+
+class BadLambda(Exception):
+    pass
+
+
+class MismatchedDomain(Exception):
+    pass
+
+
+class LayerOps:
+    def __init__(self, score=None, comment=None, enabled=None, colors=None,
+                 metadata=None, name=None, desc=None, default_values=None):
+        """
+            Initialization - configures the object to handle processing
+                based on user provided Lambdas
+            :param score: lambda to calculate score
+            :param comment: lambda to generate comments
+            :param enabled: lambda to determine enabled status
+            :param metadata: lambda to generate metadata
+            :param name: new name to apply to the resulting layer
+            :param desc: new description to apply to the resulting layer
+            :param default_values: dictionary containing desired default
+                values for missing data element values
+        """
+        self.score = score
+        self.comment = comment
+        self.enabled = enabled
+        self.colors = colors
+        self.metadata = metadata
+        self.name = name
+        self.desc = desc
+        self.default_values = {
+            "comment": "",
+            "enabled": True,
+            "color": "#ffffff",
+            "score": 1,
+            "metadata": []
+        }
```

Comment:
Shouldn't all of these be prefixed with `_` so that it's obvious they're internal? If I want to figure out the API to a given object, I'm personally inclined to use `dir`. Without the `_` prefix these appear as properties I _should_ be caring about.

```
>>> op = LayerOps(score=lambda x: sum(x))
>>> dir(op)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_applyOperation', '_build_template', '_compute', '_grabDict', '_grabElement', '_grabList', '_inDict', '_merge_to_template', '_template', 'colors', 'comment', 'default_values', 'desc', 'enabled', 'metadata', 'name', 'process', 'score']
>>>
```
When really the API is just `process`.